server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationStrategy.scala

```diff
@@ -0,0 +1,103 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.util.Base64
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.boolean.Or
+import eu.timepit.refined.collection.NonEmpty
+import eu.timepit.refined.refineV
+import eu.timepit.refined.string.MatchesRegex
+import javax.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.jmap.http.UserCredential._
+import org.apache.james.mailbox.{MailboxManager, MailboxSession}
+import org.apache.james.user.api.UsersRepository
+import org.slf4j.LoggerFactory
+import reactor.core.publisher.Mono
+import reactor.core.scala.publisher.SFlux
+import reactor.netty.http.server.HttpServerRequest
+
+import scala.compat.java8.StreamConverters._
+
+
+object UserCredential {
+  type StringNonNull = String Refined NonEmpty
+  type BasicAuthenticationHeaderValue = String Refined MatchesRegex["Basic [\\d\\w=]++"]
+  type CredentialsAsString = String Refined Or[
+  //  Support username with domain, include special characters in username and password
+    MatchesRegex["([\\w\\d\\.!#$%&*+/=?^_`{|}~-]+)@([\\w]+)([\\.]{1}[\\w]+)+:[\\w\\d\\.!#$%&*+/=?^_`{|}~-].*"],
+  //  Support username without domain, include special characters in username and password
+    MatchesRegex["[\\d\\w\\s\\S\\W].*:[\\d\\w].*"]
```

Comment:
I raise a huge warning about the complexity of these regular expression here.

I do believe we **should not** leak out the knowledge of what a user is (as a string) in this piece of code. Defining very precise REGEX here creates coupling. Saying that "we handle username with and without domain part" also creates coupling. 

This will makes refactoring harder.

I suggest we define `CredentialsAsString` as the simplest REGEX possible : `String Refined MatchesRegex[".++:.*"] `